### PR TITLE
feat(typeahead): optionally do not hide the results on blur

### PR DESCRIPTION
**Pull request by ghost**
was added:
- a property (`typeaheadIsHideOnBlur = false`) prevents hiding the results until a user doesn't choose an item in the typeahead's list.
- a demo for `typeaheadIsHideOnBlur` property
- fix an issue - closing results on dbclick on input (when on the first click on input results open and on the second click on click they hide without chosen item)
Closes to #2059

### CODE REVIEW

**Review by Domainv — APPROVED** [review]

**Review by Domainv — CHANGES_REQUESTED** [review]
Needs add a test for the functionality

File: src/spec/typeahead.directive.spec.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ class TestTypeaheadComponent {
   onBlurEvent(activeItem) { return undefined; }
 }
 
-describe('Directive: Typeahead', () => {
+fdescribe('Directive: Typeahead', () => {
```
Comment (Domainv) [inline src/spec/typeahead.directive.spec.ts]: Please remove 'fdescribe' => 'describe'

**Review by Domainv — DISMISSED** [review]

File: src/spec/typeahead.directive.spec.ts (hunk 1 above)
Comment (ghost) [reply src/spec/typeahead.directive.spec.ts]: fixed

**Comment by ludmilanesvitiy** [comment]
Typeahead Demo smoke tested in Chrome, FF - looks good.

**Comment by Domainv** [comment]
screenshot from 2018-12-12 16-15-17

File: src/typeahead/typeahead.directive.ts [hunk 2]
```diff
@@ -137,21 +141,24 @@ export class TypeaheadDirective implements OnInit, OnDestroy {
...
+    Object.assign(this, config);
   }
 
   ngOnInit(): void {
     this.typeaheadOptionsLimit = this.typeaheadOptionsLimit || 20;
     this.typeaheadMinLength =
       this.typeaheadMinLength === void 0 ? 1 : this.typeaheadMinLength;
     this.typeaheadWaitMs = this.typeaheadWaitMs || 0;
-
+    const isTypeaheadIsHideOnBlurInput = !this.typeaheadIsHideOnBlur || !this.config.typeaheadIsHideOnBlur;
```
Comment (Domainv) [inline src/typeahead/typeahead.directive.ts]: Change please to 
const showOnBlur = !this.typeaheadIsHideOnBlur || !this.config.hideOnBlur;

File: demo/src/app/components/+typeahead/typeahead-section.list.ts [hunk 3]
```diff
@@ -195,6 +196,17 @@ export const demoComponentContent: ContentSection[] = [
         component: require('!!raw-loader?lang=typescript!./demos/on-select/on-select.ts'),
         html: require('!!raw-loader?lang=markup!./demos/on-select/on-select.html'),
         outlet: DemoTypeaheadOnSelectComponent
+      },
+      {
+        title: 'Show results on blur',
+        anchor: 'show-on-blur',
+        description: `
+          <p>Use <code>typeaheadIsHideOnBlur</code> property to prevent hiding typeahead's results
```
Comment (Domainv) [inline demo/src/app/components/+typeahead/typeahead-section.list.ts]: Add description for config property 'hideOnBlur'

**Review by Domainv — DISMISSED** [review]

**Comment by ludmilanesvitiy** [comment]
Retested, looks good. RFM

**Comment by ludmilanesvitiy** [comment]
Retested, looks good. RFM

**Review by valorkin — APPROVED** [review]

**Pull request merged**